### PR TITLE
[LLM:Bugfix] Fix embedding QNN export paths

### LINKED ISSUE #4481: Qwen3-Embedding-0.6B generate_llm_qnn.py 再最新代码中报错
改bug已经被修复 [LLM:Bugfix] Fix Qwen3-Embedding QNN export crash #4216， 但后续提交覆盖了这个修改
1. generate_llm_qnn.py 脚本问题：后续提交覆盖了 commit 5889d95 的修复，使用了错误的 IO 生成流程，总是为 LLM 生成包含 logits_index 输入和 logits 输出的配置，但 embedding 模型使用的是 sentence_embeddings 输出。
2. generateLlmIO.cpp 工具问题：
   - 硬编码 llm.mnn 作为模型文件名，但 embedding 模型使用 embedding.mnn
   - isEmbeddingModel() 函数无法从 llm_config.json 中检测到 embedding 模型（因为缺少 output_names 和 model_type 字段）

### PR DESCRIPTION
**Pull request by ZedingZhang**
### Summary
- detect embedding models in the QNN export wrapper and generate sentence_embeddings IO without logits_index
- use embedding.mnn / embedding.mnn.weight for embedding model QNN export and config_qnn.json
- restore embedding exporter output prefix and make generateLlmIO select embedding.mnn with safer qwen3 detection

Fixes #4481

### Tests
- python3 -m py_compile transformers/llm/export/npu/generate_llm_qnn.py transformers/llm/export/llmexport.py
- inline Python checks for embedding and normal LLM input.json generation
- cmake -S . -B build -DMNN_BUILD_LLM=ON -DMNN_LLM_BUILD_DEMO=ON -DMNN_BUILD_TEST=OFF -DMNN_BUILD_BENCHMARK=OFF -DMNN_BUILD_DEMO=OFF -DMNN_BUILD_CONVERTER=OFF -DMNN_BUILD_TRAIN=OFF -DMNN_BUILD_QUANTOOLS=OFF
- cmake --build build --target generateLlmIO -j4

### CODE REVIEW

**Comment by ZedingZhang** [comment]
Closing for now to run local Qwen3-Embedding-0.6B export validation before reopening.

**Comment by ZedingZhang** [comment]
Local validation completed for this PR: Qwen3-Embedding-0.6B QNN export passed in a linux/amd64 Docker container with QAIRT/QNN SDK 2.47.0.260601. Verified outputs include qnn/embedding.mnn, embedding.mnn.weight, and 29 graph*.bin context binaries. config_qnn.json points to qnn/embedding.mnn and embedding.mnn.weight with chunk_limits [128,1].

**Comment by ZedingZhang** [comment]
@wangzhaode @Qxinyu I have rebased this PR onto the latest master and resolved the conflicts. The QNN embedding export configuration has been validated locally. Could you please take a look when you have a chance? Thank you!

**Review by wangzhaode — APPROVED** [review]
